### PR TITLE
Add credentials endpoint option for awslogs driver

### PR DESCRIPTION
**Pull request by adnxn**
/cc @samuelkarp 

**- What I did**
I added the awslogs driver option `awslogs-credentials-endpoint` to use relative URI paths to retrieve credentials from the aws sdk's default remote endpoint. The default remote endpoint for the sdk is at `http://169.254.170.2`. Otherwise, credentials from EC2 Instance Metadata will be used to send log events to cloudwatch logs. This new option would allow users to override that behavior, where they can specify a different relative uri path such as `/v1/creds` for the container's cloudwatch credentials.

**- How I did it**
Added `credentialsEndpointKey` to retrieve and consume the relative path and use that as the credentials provider endpoint.

**- How to verify it**
I added a test case to verifiy that the awslogs driver will retrieve credentials from a http endpoint. I've also added tests for the existing ways the awslogs driver can fetch credentials. It can be also be manually verified by starting a container with the extra `--log-opt awslogs-credentials-endpoint=/v1/creds`, assuming there is a http credential provider serving credentials from `http://169.254.170.2/v1/creds`

**- Description for the changelog**

awslogs driver can now fetch credentials from user defined relative URI endpoint.

### CODE REVIEW

File: daemon/logger/awslogs/cloudwatchlogs_test.go [hunk 1]
```diff
@@ -1065,3 +1068,115 @@ func BenchmarkUnwrapEvents(b *testing.B) {
...
+	// set the SDKEndpoint in the driver
+	newSDKEndpoint = testServer.URL
+
+	info := logger.Info{
+		Config: map[string]string{},
+	}
+
+	info.Config["awslogs-credentials-endpoint"] = "/creds"
+
+	c, err := newAWSLogsClient(info)
```
Comment (samuelkarp) [inline daemon/logger/awslogs/cloudwatchlogs_test.go]: `err` is never read.  You either want to use `_` or actually check it.  This also appears on lines 1128 and 1174.

> daemon/logger/awslogs/cloudwatchlogs_test.go:1100:5:warning: ineffectual assignment to err (ineffassign)

**Review by samuelkarp — APPROVED** [review]
LGTM after fixing the ineffassign issues.

**Comment by adnxn** [comment]
z — Jenkins build Docker-PRs-s390x 6041 has failed. This is unrelated to changes made in the PR, specifically `DockerSuite.TestUpdateKernelMemoryUninitialized` in `docker_cli_update_unix_test.go` is failing.

**Comment by yongtang** [comment]
z failure is related to #34938

File: daemon/logger/awslogs/cloudwatchlogs_test.go (hunk 1 above)
Comment (adnxn) [reply daemon/logger/awslogs/cloudwatchlogs_test.go]: fixed.

**Comment by adnxn** [comment]
cc @cpuguy83 PTAL

**Review by yongtang — APPROVED** [review]
LGTM

**Comment by crosbymichael** [comment]
LGTM

**Comment by thaJeztah** [comment]
@adnxn Thanks! Can you open a documentation pull-request against the `vnext-engine` branch in the docs repository? https://github.com/docker/docker.github.io/blob/vnext-engine/engine/admin/logging/awslogs.md

**Pull request merged**